spark/src/main/scala/org/apache/spark/sql/delta/util/TimestampFormatter.scala

```diff
@@ -67,15 +67,15 @@ sealed trait TimestampFormatter extends Serializable {
 
 class Iso8601TimestampFormatter(
     pattern: String,
-    timeZone: TimeZone,
+    timeZone: ZoneId,
```

Comment:
I don't think changing this poses a compatibility concern as long as the various apply methods still are compatible (My scala is a bit rusty so let me know, but I'm unable to initialize this outside anyways)?

As for why the change:

Spark SQL supports setting any zoneID like "UTC-08:00" etc. However, if one tries to convert this into java timezone type via TimeZone.of(ZoneId.of("UTC-08:00")) then the timezone actualy ends up being just "UTC" because timezone API can't handle the offset -08:00. So passing a zoneID avoids a correctness issue there.